internal/locate/region_request.go

```diff
@@ -1202,6 +1216,10 @@ func (s *RegionRequestSender) sendReqToRegion(bo *retry.Backoffer, rpcCtx *RPCCo
 	if !injectFailOnSend {
 		start := time.Now()
 		resp, err = s.client.SendRequest(ctx, sendToAddr, req, timeout)
+		// Record timecost of this request into the related Store.
+		if !util.IsInternalRequest(req.RequestSource) {
+			rpcCtx.Store.recordSlowScoreStat(time.Since(start))
```

Comment:
is this called when preferleader mode is off?

Reply:
Yes, this condition is just filtering out unnecessary internal requests sent to TiKV node by ticking, making Client focus on the timecost of external requests, triggered by users.